### PR TITLE
[web] Honor background color when rendering text to dom

### PR DESCRIPTION
**Pull request by mdebbar**
Goldens PR: https://github.com/flutter/goldens/pull/207

Fixes https://github.com/flutter/flutter/issues/84763

### CODE REVIEW

File: lib/web_ui/lib/src/engine/text/paragraph.dart [hunk 1]
```diff
@@ -1745,6 +1745,10 @@ void _applyTextStyleToElement({
   if (color != null) {
     cssStyle.color = colorToCssString(color);
   }
+  final ui.Color? background = style._background?.color;
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/text/paragraph.dart:1748]: Totally unimplemented? Yikes! :P

**Review by ditman — APPROVED** [review]
LGTM!

File: lib/web_ui/lib/src/engine/text/paragraph.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/text/paragraph.dart:1748]: Long story :P

In the old implementation, we used to only support plain text (with very minimal rich text support). If the paragraph has a red background for example, we used to just draw a red rectangle behind the paragraph.

With the recent rich text implementation, we started drawing rectangles for each span that has a background (in canvas2d mode). But in DOM mode, we reused some of the old code, which didn't do backgrounds for spans.

Comment (ditman) [reply lib/web_ui/lib/src/engine/text/paragraph.dart:1748]: Ahhh, yes, I remember the rectangle drawing part!

**Pull request merged**